### ISSUE TITLE
[Mail Alerts] could use remote SMTP with optional authentication 

### ISSUE THREAD

**Issue by UltimateByte** [GameServerManagers/LinuxGSM#1690]
Sending proper emails require a lot of configuration, and it's very unlikely that a game server would be properly set up for that purpose. 

In order to prevent mail notifications from falling right into the junk dir, it would be nice to allow for proper authentication to a remote server. The default config would still use the local server to send emails, but would allow able to send mails from a remote server, with or without authentication.

Here is an overview of the minimum required config variables to send an authenticated email from a local or remote server:
```bash
# Required settings
emailalert="on/off"
mailfrom="${servicename}@localhost" # Who is the sender address
mailto="youradmin@address.com" # Who is the recipient
# Optional settings
mailserver="localhost" # Which SMTP server to use
mailcypher="on/off" # Wether to use SSL/TLS connection or not
mailport="25" # Which port to use, 25 is unsecured standard, 465 & 587 are SSL/TLS ports
mailauthentication="on/off" # Do your mail server require authentication
mailauthuser="root@localhost" # What email address to log into (required if authentication is  on)
mailauthpassword="yourpassword" # Your email password (required if authentication is on)
```

Now I'm not sure yet how to send emails with these settings from bash, but I'm confident that there are easy ways to send authenticated emails.

**Comment by msgerbs** [comment]
Don't forget about secure SMTP settings.

**Comment by UltimateByte** [comment]
Already thought about it :p 

`mailcypher="on/off" # Wether to use SSL/TLS connection or not`

**Comment by JimTR** [comment]
Use heirloom mailx
after installed 
try reading http://www.binarytides.com/linux-mailx-command/ I have been using this for years and it works

**Comment by UltimateByte** [comment]
Awesome @JimTR it seems to be exactly what we needed to make the implementation easy! Thanks!

**Comment by UltimateByte** [comment]
I couldn't get it to work using mailx because `-S` argument is unknown on my distro (Debian 9). I tried with mailx, `heirloom-mailx` and `bsd-mailx`: none worked.

However, I successfuly achieved this using `sendemail`.

Command looks like this
`sendemail -m -o tls=no -s mail.mydomain.fr:25 -f "${emailfrom}" -t "${email}" -u "${alertsubject}" -xu "${emailfrom}" -xp "${emailpassword}" -m "$(cat ${alertlog})"`

However, SSL/TLS/STARTTLS won't work by default.

TLS Enabled on port 465:
```
[ .... ] Alert gmodserver: Sending Email alert: admin@mydomain.tld May 14 20:07:57 gaben sendemail[14634]: ERROR => No TLS support!  SendEmail can't load required libraries. (try installing Net::SSLeay and IO::Socket::SSL)
[ FAIL ] Alert gmodserver: Sending Email alert: admin@mydomain.tld
```

Same on port 587 (expecting STARTTLS).

After installing a few packages, I've been able to connect using STARTTLS on port 587 with `tls=yes` argument, which means like many, they're calling "TLS" what is actually STARTTLS.

`apt install libnet-ssleay-perl libio-socket-ssl-perl`
